### PR TITLE
Call rm_ensure_result() in rm_set_user_artifact()

### PR DESCRIPTION
**Pull request by Watson1978**

### CODE REVIEW

File: ext/RMagick/rmutil.c [hunk 1]
```diff
@@ -1005,6 +1005,8 @@ void rm_set_user_artifact(Image *images, Info *info)
 {
     const char *value;
 
+    rm_ensure_result(images);
```
Comment (dlemstra) [inline ext/RMagick/rmutil.c]: I really don't think that this is the place to check this. You are now introducing a side-effect in this method call. Can we revert this change?

cc @mockdeep

Comment (mockdeep) [reply ext/RMagick/rmutil.c]: @dlemstra @Watson1978 what do `rm_ensure_result` and `rm_set_user_artifact` do?

Comment (Watson1978) [reply ext/RMagick/rmutil.c]: Make sense. It certainly has side effects.

`rm_ensure_result` raise a exception if `images` is `NULL`. However, `rm_set_user_artifact` was just skip it.
